src/ngraph/graph_util.cpp

```diff
@@ -46,17 +46,29 @@ void ngraph::traverse_nodes(const std::shared_ptr<const Function> p,
 
 void ngraph::traverse_nodes(const Function* p, std::function<void(std::shared_ptr<Node>)> f)
```

Comment:
Can you put as a comment whta your wrote in the description of this PR? Also is there a way you can add a unit test so that the next time someone makes a change it does not again produce unsatisfactory results?